src/search/findSimilarNotes.ts

```diff
@@ -0,0 +1,190 @@
+import { DBOperations } from "@/search/dbOperations";
+import { getSettings } from "@/settings/model";
+import { extractNoteTitles, getNoteFileFromTitle } from "@/utils";
+import { InternalTypedDocument, Orama, Result } from "@orama/orama";
+import { TFile } from "obsidian";
+
+const MIN_SIMILARITY_SCORE = 0.3;
+const MAX_K = 20;
+const ORIGINAL_WEIGHT = 0.7;
+const LINKED_WEIGHT = 0.3;
+
+/**
+ * Checks if the index exists for the given file path.
+ * @param db - The Orama database.
+ * @param filePath - The file path to check.
+ * @returns True if the index exists, false otherwise.
+ */
+async function checkIndexExists(db: Orama<any>, filePath: string) {
+  const hits = await DBOperations.getDocsByPath(db, filePath);
+  if (!hits) {
+    return false;
+  }
+  // Filter the results to only include exact path matches
+  const exactMatches = hits.filter((hit) => hit.document.path === filePath);
+  return exactMatches.length > 0 && exactMatches.some((hit) => hit.document.embedding);
+}
+
+/**
+ * Gets the embeddings for the given note titles.
+ * @param noteTitles - The note titles to get embeddings for.
+ * @param db - The Orama database.
+ * @returns The embeddings for the given note titles.
+ */
+async function getEmbeddings(noteTitles: string[], db: Orama<any>): Promise<number[][]> {
+  const debug = getSettings().debug;
+  const embeddings: number[][] = [];
+  for (const noteTitle of noteTitles) {
+    const noteFile = await getNoteFileFromTitle(app.vault, noteTitle);
```

Comment:
Not particularly for this PR but right now we are not differentiating notes with the same title under different paths. This was on me. In the future we should use a unique identifier i.e. note path.

When a user types `[[` the list should show the corresponding path next to each title, and in the background only path should be used as the unique id.